modP2pImpl/src/org/aion/p2p/impl1/P2pMgr.java

```diff
@@ -577,7 +130,7 @@ public P2pMgr(int _netId, String _revision, String _nodeId, String _ip, int _por
 
         for (String _bootNode : _bootNodes) {
             Node node = Node.parseP2p(_bootNode);
-            if (node != null && validateNode(node)) {
+            if (validateNode(node)) {
```

Comment:
Minor: Any reason why the null check is removed? If the first condition is false, validateNode is not called (&& is short circuited).

Reply:
I did this because validateNode immediately returns false when its input is null, seemed to me like an unnecessary duplication of a null check

Reply:
I believe validateNode will not be called when node is null.
 https://stackoverflow.com/questions/8759868/java-logical-operator-short-circuiting

Reply:
That’s true, sorry I didn’t mean the check is duplicated, I meant the validate function handles the null case already so the explicit null check before it isn’t needed for correctness. But you’re right no need to enter the function and grow the call stack etc.